packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -255,70 +291,92 @@ const ProductDetail = () => {
+                                    : handleProductBundleAddToCart
+                            }
                             addToWishlist={handleAddToWishlist}
                             isProductLoading={isProductLoading}
                             isWishlistLoading={isWishlistLoading}
-                            validateOrderability={handleProductSetValidation}
+                            validateOrderability={handleChildProductValidation}
+                            childProductOrderability={childProductOrderability}
                         />
 
                         <hr />
 
                         {/* TODO: consider `childProduct.belongsToSet` */}
                         {
-                            // Product Set: render the child products
-                            product.setProducts.map((childProduct) => (
-                                <Box key={childProduct.id} data-testid="child-product">
-                                    <ProductView
-                                        // Do no use an arrow function as we are manipulating the functions scope.
-                                        ref={function (ref) {
-                                            // Assign the "set" scope of the ref, this is how we access the internal
-                                            // validation.
-                                            childProductRefs.current[childProduct.id] = {
-                                                ref,
-                                                validateOrderability: this.validateOrderability
+                            // Render the child products
+                            normalizedProduct.childProducts.map(
+                                ({product: childProduct, quantity: childQuantity}) => (
+                                    <Box key={childProduct.id} data-testid="child-product">
+                                        <ProductView
+                                            // Do not use an arrow function as we are manipulating the functions scope.
+                                            ref={function (ref) {
+                                                // Assign the "set" scope of the ref, this is how we access the internal
+                                                // validation.
+                                                childProductRefs.current[childProduct.id] = {
+                                                    ref,
+                                                    validateOrderability: this.validateOrderability
+                                                }
+                                            }}
+                                            product={childProduct}
+                                            isProductPartOfSet={isProductASet}
+                                            isProductPartOfBundle={isProductABundle}
+                                            bundleQuantity={childQuantity}
```

Comment:
I think `bundleQuantity` is not accurate name for what it really is. It's not the quantity of the bundle, but the quantity of that particular child in the bundle.

How about giving it a more generic name like `productQuantity`? An optional value.. if it's defined, then it will be rendered.

Reply:
renamed to `childOfBundleQuantity`